builtin/merge-ours.c

```diff
@@ -11,12 +11,24 @@
 #include "git-compat-util.h"
 #include "builtin.h"
 #include "diff.h"
+#include "parse-options.h"
+
+static const char * const merge_ours_usage[] = {
+	N_("git merge_ours [arg]..."),
+	NULL,
+};
 
 static const char builtin_merge_ours_usage[] =
 	"git merge-ours <base>... -- HEAD <remote>...";
```

Comment:
To respond to dscho's comment above, there's in fact a previous usage string here (which says pretty much what dscho suggested). I expect you will catch some sideways glances on list for changing the usage text, so in my opinion you had better adopt this usage string.

The task at hand for the microproject is not to add parse-options, so much as to replace the manual option parsing with parse-options. So, throughout this file after your change, there is duplicated effort - first you do it with parse-options.h (the right way, per the project description) and then you do it with manual argv parsing (the wrong way, per the project description).

Instead, you should try to understand what the code is doing now, and replace it with the parse-options equivalent. So this line says "Define the usage string as "git merge-ours blah blah blah""; above, you've added the line where you define the usage string, which means this line is now redundant. I see similar through the rest of the code (and dscho commented on some).

Reply:
Happily, `git-merge-ours` does not take any options, so you'll just need to remove that "other" usage, and remove the check for `-h`.

Reply:
I'm trying to push my edit but I keep getting this error :(

To https://github.com/Georgecanfly/git.git
 ! [rejected]              merge-ours -> merge-ours (non-fast-forward)
error: failed to push some refs to 'https://github.com/Georgecanfly/git.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

I looked up that error and found a solution would be using 
git reset --hard origin/branch-name

I did that to every branch but I have had no luck.

Reply:
You need to force-push the change, but please note that most likely, your `git reset --hard` invocations completely undid your local changes. You might want to look through `git log -g` to find the commit to which you _actually_ want to reset, and then you can force-push using `git push --force-with-lease <remote> HEAD` (where `<remote>` is the nickname for the remote you added that points to your fork of git.git).

Reply:
George, the relevant section of the tutorial is here: https://git-scm.com/docs/MyFirstContribution.html#responding-ggg where it explains you do need to force-push to your branch with updates like this.

Reply:
Oh great! Thank you so much.

Reply:
Oh dear, from the current diff, it does not look like the correct commit made it through; now it does not use `parse_options()` at all anymore...

Reply:
I think I fixed everything now @dscho Thank you and for the help!!!